CMakeLists.txt

```diff
@@ -280,12 +280,23 @@ add_subdirectory(library)
 # to define the test executables.
 #
 if(ENABLE_TESTING OR ENABLE_PROGRAMS)
-    file(GLOB MBEDTLS_TEST_FILES ${CMAKE_CURRENT_SOURCE_DIR}/tests/src/*.c ${CMAKE_CURRENT_SOURCE_DIR}/tests/src/drivers/*.c)
+    file(GLOB MBEDTLS_TEST_FILES
+         ${CMAKE_CURRENT_SOURCE_DIR}/tests/src/*.c
+         ${CMAKE_CURRENT_SOURCE_DIR}/tests/src/drivers/*.c)
     add_library(mbedtls_test OBJECT ${MBEDTLS_TEST_FILES})
     target_include_directories(mbedtls_test
         PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/tests/include
         PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include
         PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/library)
+
+    file(GLOB MBEDTLS_TEST_HELPER_FILES
```

Comment:
I would like to document a rule: if ??? then you must put your test helpers in the `test_helpers` subdirectory. But I have no idea what ??? should be.

Reply:
I think it is just that the crypto programs link with the "mbedtls_test" objects (test/src/*) and only the crypto library, not the TLS one. Thus if in "mbedtls_test" objects there are references to TLS functions, they are not resolved.

Reply:
Ah, that would be consistent with that I've seen so far. So `test_helpers/` should actually be `ssl_helpers/`?

Reply:
That would be better probably and maybe there is/will be a need for x509_helpers as well.

Reply:
> I think it is just that the crypto programs link with the "mbedtls_test" objects (test/src/*) and only the crypto library, not the TLS one. Thus if in "mbedtls_test" objects there are references to TLS functions, they are not resolved.

There are maybe other problems, I stopped after finding the above. Otherwise a crypto program should not call any TLS helper thus we could wonder why it does not work anyway.